org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend

```diff
@@ -409,6 +464,155 @@ class CapabilitiesEventArguments {
+	 * <p>
+	 * This is an optional property.
+	 */
+	Double percentage;
+}
+
+/**
+ * The event signals that the progress reporting needs to updated with a new message and/or percentage.
+ * <p>
+ * The client does not have to update the UI immediately, but the clients needs to keep track of the message
+ * and/or percentage values.
+ * <p>
+ * This event should only be sent if the client has passed the value true for the 'supportsProgressReporting'
+ * capability of the 'initialize' request.
+ */
+@JsonRpcData
+class ProgressUpdateEventArguments {
+	/**
+	 * The ID that was introduced in the initial 'progressStart' event.
+	 */
+	@NonNull
+	String progressId;
+	/**
+	 * Optional, more detailed progress message. If omitted, the previous message (if any) is used.
+	 * <p>
+	 * This is an optional property.
+	 */
+	String message;
+	/**
+	 * Optional progress percentage to display (value range: 0 to 100). If omitted no percentage will be shown.
+	 * <p>
+	 * This is an optional property.
+	 */
+	Double percentage;
+}
+
+/**
+ * The event signals the end of the progress reporting with an optional final message.
+ * <p>
+ * This event should only be sent if the client has passed the value true for the 'supportsProgressReporting'
+ * capability of the 'initialize' request.
+ */
+@JsonRpcData
+class ProgressEndEventArguments {
+	/**
+	 * The ID that was introduced in the initial 'ProgressStartEvent'.
+	 */
+	@NonNull
+	String progressId;
+	/**
+	 * Optional, more detailed progress message. If omitted, the previous message (if any) is used.
+	 * <p>
+	 * This is an optional property.
+	 */
+	String message;
+}
+
+/**
+ * This event signals that some state in the debug adapter has changed and requires that the client needs to
+ * re-render the data snapshot previously requested.
+ * <p>
+ * Debug adapters do not have to emit this event for runtime changes like stopped or thread events because in that
+ * case the client refetches the new state anyway. But the event can be used for example to refresh the UI after
+ * rendering formatting has changed in the debug adapter.
+ * <p>
+ * This event should only be sent if the debug adapter has received a value true for the
+ * 'supportsInvalidatedEvent' capability of the 'initialize' request.
+ */
+@JsonRpcData
+class InvalidatedEventArguments {
+	/**
+	 * Optional set of logical areas that got invalidated. This property has a hint characteristic: a client can only
+	 * be expected to make a 'best effort' in honouring the areas but there are no guarantees. If this property is
+	 * missing, empty, or if values are not understand the client should assume a single value 'all'.
+	 * <p>
+	 * This is an optional property.
+	 * <p>
+	 * Possible values include - but not limited to those defined in {@link InvalidatedAreas}
+	 */
+	String[] areas;
+	/**
+	 * If specified, the client only needs to refetch data related to this thread.
+	 * <p>
+	 * This is an optional property.
+	 */
+	Double threadId;
```

Comment:
This is wrong see https://github.com/microsoft/debug-adapter-protocol/issues/145

Reply:
Subsequent commit to this branch has `threadId` and `frameId` as an `Integer`